### PR TITLE
Feature/163153545/hash store password

### PR DESCRIPTION
**Pull request by rajeman**
#### What does this PR do?
- It encrypts passwords before storing in User model
#### Description of Task to be completed?
- [x] add a <code>beforeCreate</code> hook to User model to hash password before storage
#### How should this be manually tested?
- send a valid post request to <code>/users</code> and check the password column of the User table in the database for the newly added user
- alternatively you can run <code>npm test</code> and check the test database
- you should see an hash in place of a plain string
#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
[#163153545](https://www.pivotaltracker.com/n/projects/2229853/stories/163153545)
#### Screenshots (if appropriate)
image

### CODE REVIEW

File: models/user.js [hunk 1]
```diff
@@ -20,7 +22,15 @@ module.exports = (sequelize, DataTypes) => {
       type: DataTypes.BOOLEAN,
       defaultValue: false
     },
-  });
+  }, {
+    hooks: {
+      beforeCreate: (user) => {
+        const salt = bcrypt.genSaltSync(10);
```
Comment (oreoluwade) [inline models/user.js]: This is good... However, when we need to authenticate the user, we'd need to check that the password is valid. To this, end, It might be useful to extract lines 28 and 29 into a method on the prototype of the `User` model/class and then create another method on the prototype for checking validity of the password.

Picture the prototype method as a method of a class that can be called on all objects created from the class. This is why we are hashing the password on the User model in the first place so that all users created have access to those methods.

TLDR, what you'd do will bear semblance to:

      `hooks: {
        beforeCreate(user) {
          user.hashPassword();
        }
      }`

And then Above `return User` in the model, you have these two functions. I put dummies below:

    `User.prototype.hashPassword = function hashPassword() {
          return......;
      };  `

      User.prototype.validPassword = function validPassword(password) {
        return ....;
      };

Comment (oreoluwade) [reply models/user.js]: Check the `to hash a password` and `to check a password` sections of the `bcrypt` npm module to see what could be the contents of the two prototype methods above.

Comment (rajeman) [reply models/user.js]: I will work on it

File: models/user.js [hunk 2]
```diff
@@ -20,10 +22,28 @@ module.exports = (sequelize, DataTypes) => {
...
+  );
+
   User.associate = models => User.hasOne(models.Profile, {
     foreignKey: 'userId',
     onDelete: 'CASCADE'
   });
+
+  User.prototype.hashPassword = async function hashPassword() {
+    const saltRounds = 10;
+    this.password = await bcrypt.hash(this.password, saltRounds);
```
Comment (oreoluwade) [inline models/user.js:41]: Good job.

Just return `this.password` here after hashing, and then we are set.

**Review by oreoluwade — APPROVED** [review]

File: models/user.js (hunk 2 above)
Comment (rajeman) [reply models/user.js:41]: Alright
